code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ProductRatingNotificationBuilderTest.kt

```diff
@@ -0,0 +1,190 @@
+        every { cacheImages(any()) } answers { 0 }
+        every { any<RemoteViews>().setRemoteViewImage(any(), any()) } returns true
+        every { anyConstructed<RemoteViews>().setViewVisibility(any(), View.GONE) } just Runs
+
+        val pushTemplate = provideMockedProductRatingTemplate()
+        val notificationBuilder = ProductRatingNotificationBuilder.construct(context, pushTemplate, trackerActivityClass, broadcastReceiverClass)
+
+        verify(exactly = 1) { anyConstructed<RemoteViews>().setViewVisibility(R.id.expanded_template_image, View.GONE) }
+    }
+
+    @Test
+    fun `construct should set image for expanded layout if image is downloaded successfully`() {
+        val cachedItem = mockkClass(Bitmap::class)
+
+        every { cacheImages(any()) } answers { 1 }
+        every { getCachedImage(any()) } answers { cachedItem }
+        every { any<RemoteViews>().setRemoteViewImage(any(), any()) } returns true
+        every { anyConstructed<RemoteViews>().setViewVisibility(any(), View.GONE) } just Runs
+
+        val pushTemplate = provideMockedProductRatingTemplate()
+        val notificationBuilder = ProductRatingNotificationBuilder.construct(context, pushTemplate, trackerActivityClass, broadcastReceiverClass)
+
+        verify(exactly = 1) { anyConstructed<RemoteViews>().setImageViewBitmap(R.id.expanded_template_image, any()) }
+    }
+
+    @Test
+    fun `Rating Confirmation action should be hidden if no rating has been selected`() {
+        every { any<RemoteViews>().setRemoteViewImage(any(), any()) } returns true
+        every { anyConstructed<RemoteViews>().setViewVisibility(any(), any()) } just Runs
+
+        val pushTemplate = provideMockedProductRatingTemplate()
+        val notificationBuilder = ProductRatingNotificationBuilder.construct(context, pushTemplate, trackerActivityClass, broadcastReceiverClass)
+
+        verify(exactly = 1) { anyConstructed<RemoteViews>().setViewVisibility(R.id.rating_confirm, View.INVISIBLE) }
+    }
+
+    @Test
+    fun `Rating confirmation action should be visible when rating is selected`() {
+        every { any<RemoteViews>().setRemoteViewImage(any(), any()) } returns true
+        every { anyConstructed<RemoteViews>().setViewVisibility(any(), any()) } just Runs
+
+        val dataMap = getMockedDataMapForRatingTemplate()
+        dataMap[PushTemplateConstants.IntentKeys.RATING_SELECTED] = "3"
+        val pushTemplate = ProductRatingPushTemplate(MapData(dataMap))
+        val notificationBuilder = ProductRatingNotificationBuilder.construct(context, pushTemplate, trackerActivityClass, broadcastReceiverClass)
+
+        verify(exactly = 1) { anyConstructed<RemoteViews>().setViewVisibility(R.id.rating_confirm, View.VISIBLE) }
+    }
+
+    @Test
+    fun `construct should throw NotificationConstructionFailedException if image for unselected rating icon is invalid`() {
+        val pushTemplate = provideMockedProductRatingTemplate()
```

Comment:
Are we missing `every { any<RemoteViews>().setRemoteViewImage("rating_star_outline", any()) } returns false` here?